diff --git a/src/logid/CMakeLists.txt b/src/logid/CMakeLists.txt
index b2616c8..c62efb7 100644
--- a/src/logid/CMakeLists.txt
+++ b/src/logid/CMakeLists.txt
@@ -109,12 +109,3 @@ if (SYSTEMD_FOUND)
 elseif (NOT SYSTEMD_FOUND AND SYSTEMD_SERVICES_INSTALL_DIR)
     message(FATAL_ERROR "systemd is not found w/ pkg-config but SYSTEMD_SERVICES_INSTALL_DIR is defined.")
 endif ()
-
-# Install DBus conf
-# TODO: Is there a better way of setting the system policy directory?
-set(DBUS_SYSTEM_POLICY_INSTALL_DIR "/usr/share/dbus-1/system.d")
-configure_file(logiops-dbus.conf.in ${CMAKE_BINARY_DIR}/pizza.pixl.LogiOps.conf)
-message(STATUS "dbus system policy will be installed at ${DBUS_SYSTEM_POLICY_INSTALL_DIR}")
-install(FILES ${CMAKE_BINARY_DIR}/pizza.pixl.LogiOps.conf
-        DESTINATION ${DBUS_SYSTEM_POLICY_INSTALL_DIR}
-        COMPONENT cp)
