diff --git a/t_modules/t_main.py b/t_modules/t_main.py
index ec02154..1472729 100644
--- a/t_modules/t_main.py
+++ b/t_modules/t_main.py
@@ -8558,7 +8558,7 @@ class SubsonicService:
                     nt.fullpath = song["path"]
                     nt.parent_folder_path = os.path.dirname(song["path"])
                 if "coverArt" in song:
-                    nt.art_url_key = song["id"]
+                    nt.art_url_key = song["coverArt"]
                 nt.url_key = song["id"]
                 nt.misc["subsonic-folder-id"] = folder_id
                 nt.is_network = True
